client/src/pages/BlogsPage.jsx

```diff
@@ -1,31 +1,37 @@
-import React, { useContext } from 'react';
-import { BlogContext } from '../context/BlogContext';
-import Blog from '../components/Blog';
-// import SearchBox from '../components/SearchBox';
-import Loader from '../components/Loader'
-import WriteButton from '../components/WriteButton'
+import { useSelector, useDispatch } from "react-redux";
+import { getAllBlogs } from "../features/blogs/blogsSlice";
+import Blog from "../components/Blog";
+import Loader from "../components/Loader";
+import WriteBlogButton from "../components/WriteBlogButton";
+import { useEffect } from "react";
 
 const BlogsPage = () => {
-  const { blogs } = useContext(BlogContext);
-  if (blogs.length === 0) {
-    return <Loader />
+  const dispatch = useDispatch();
+  const { blogs, status } = useSelector((state) => state.blogs);
+
+  useEffect(() => {
+    if (status === "idle" || blogs.length === 0) {
+      dispatch(getAllBlogs());
+    }
+  }, [status, blogs.length, dispatch]);
+
+  if (status === "loading" && !blogs) {
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider simplifying loading state check condition</b></div>
<div id="fix">

The loading check condition `status === "loading" && !blogs` may not correctly handle the initial loading state. Consider checking `status === "loading"` only since `blogs` could be an empty array rather than null/undefined.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
  if (status === "loading") {
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Divyanshu9822/blog-mern-app/pull/23#issuecomment-2544589025>#e379ba</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged